fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/AgilityContext.java

```diff
@@ -65,6 +70,42 @@ default void clear(Range range) {
         accept(context -> context.ensureActive().clear(range));
     }
 
+    default CompletableFuture<List<KeyValue>> getRange(byte[] begin, byte[] end) {
+        return apply(context -> context.ensureActive().getRange(begin, end).asList());
+    }
+
+    @Nonnull
+    FDBRecordContext getCallerContext();
```

Comment:
Perhaps worth a javadoc on this method, specifically calling out that you shouldn't actually use it.
Or perhaps it would be better to just change all the default methods below to not be default, further encapsulating the caller context.

Reply:
For now, at least, it is required for `FDBDirectory.getCallerContext`. But we can change it in the future. 
I'll add javadoc.

Reply:
✅